citation-matching/citation-matching-core-code/src/main/java/pl/edu/icm/coansys/citations/BestMatchedCitationPicker.java

```diff
@@ -0,0 +1,116 @@
+package pl.edu.icm.coansys.citations;
+
+import java.io.Serializable;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.hadoop.io.Text;
+import org.apache.spark.api.java.JavaPairRDD;
+
+import com.google.common.collect.Lists;
+
+import pl.edu.icm.coansys.citations.data.MatchableEntity;
+import pl.edu.icm.coansys.citations.data.SimilarityMeasurer;
+import pl.edu.icm.coansys.citations.data.TextWithBytesWritable;
+import scala.Tuple2;
+
+/**
+ * Picker of best matched citation among (citation, document) pairs.
```

Comment:
Well, maybe.
But in this class 'matched citation' means something more than 'citDocPair'.
Matched citation contains not only citation and document, but also similarity.